File: backend/src/main/java/com/snackoverflow/toolgether/domain/deposit/repository/DepositHistoryRepository.java
package com.snackoverflow.toolgether.domain.deposit.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.snackoverflow.toolgether.domain.deposit.entity.DepositHistory;

public interface DepositHistoryRepository extends JpaRepository<DepositHistory, Long> {
	Optional<DepositHistory> findByReservationId(Long reservationId);
}

File: backend/src/main/java/com/snackoverflow/toolgether/domain/deposit/repository/DepositHistoryRepository.kt
package com.snackoverflow.toolgether.domain.deposit.repository

import com.snackoverflow.toolgether.domain.deposit.entity.DepositHistory
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface DepositHistoryRepository : JpaRepository<DepositHistory?, Long?> {
    fun findByReservationId(reservationId: Long): Optional<DepositHistory>
}